src/goSuggest.ts

```diff
@@ -153,5 +210,58 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 				});
 			});
 		});
+		return Promise.all([pkgPromise, configPromise]).then(() => {
+			return Promise.resolve();
+		}); ;
+	}
+
+	// Return importable packages that match given word as Completion Items
+	private getMatchingPackages(word: string): vscode.CompletionItem[] {
+		if (!word) return [];
+		let completionItems = this.pkgsList.filter((pkgInfo: PackageInfo) => {
+			return pkgInfo.name.startsWith(word);
+		}).map((pkgInfo: PackageInfo) => {
+			let item = new vscode.CompletionItem(pkgInfo.name, vscode.CompletionItemKind.Keyword);
+			item.detail = 'Add import';
+			item.insertText = pkgInfo.name;
+			item.command = {
+				title: 'Import Package',
+				command: 'go.import.add',
+				arguments: [pkgInfo.path]
+			};
+			return item;
+		});
+		return completionItems;
+	}
+
+	// Given a line ending with dot, return the word preceeding the dot if it is a package name that can be imported
+	private getPackagePathFromLine(line: string): string {
+		let pkgName = null;
+
+		// There could be multiple dots in the line
+		// we are interested in the word preceeding the last one
+		let splits = line.split('.');
+		line = splits[splits.length - 2];
+
+		// There could be multiple words in the line
+		// we are interested in the last one
+		let wordmatches = null;
+		let pattern = /(\w+)/g;
```

Comment:
Also - to avoid the while loop - can't you bind it to match at the end of the line?  In fact, can't you avoid most of this logic by just using the regexp to match the `.$` at the end of the line that you are sure is there?